charmcraft/commands/pack.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+import logging
+import os
+import pathlib
+import zipfile
+
+from charmcraft.cmdbase import BaseCommand, CommandError
+from .utils import load_yaml
+
+logger = logging.getLogger(__name__)
+
+# the minimum set of files in a bundle
+MANDATORY_FILES = {'bundle.yaml'}
+
+
+def build_zip(zippath, basedir, fpaths):
+    """Build the final file.
+
+    Note we convert all paths to str to support Python 3.5.
```

Comment:
If shipping `charmcraft` a snap, why is python 3.5 required?

Reply:
Because it's also provided as something installable from PyPI (and you can run it from the project, anyway). In those cases, if you're in a Xenial machine (because you're building for Xenial), it needs to work with 3.5.

Reply:
Once the lifecycle is integrated, you will undoubtedly have a higher python requirement. It is worth considering the option of taking the position that the snap must be used if the environment does not support the minimum required versions.